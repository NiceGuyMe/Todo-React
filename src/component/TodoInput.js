import { useState } from "react";

export function TodoInput({ onAddTodo }) {
  const [todo, setTodo] = useState("");

  const handleChange = (event) => {
    setTodo(event.target.value);
  };

  const handleKeyDown = (e) => {
    if (e.key === "Enter") {
      onAddTodo(todo);
      setTodo("");
    }
  };

  return (
    <input
      placeholder="Enter todo"
      className="input"
      type="text"
      value={todo}
      onChange={handleChange}
      onKeyDown={handleKeyDown}
    />
  );
}
