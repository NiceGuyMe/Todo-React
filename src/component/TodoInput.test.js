import React from "react";
import { TodoInput } from "./TodoInput";
import renderer from "react-test-renderer";

it("rend correctement", () => {
  const tree = renderer.create(<TodoInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
