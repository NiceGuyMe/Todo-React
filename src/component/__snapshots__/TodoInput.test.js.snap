// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctement 1`] = `
<input
  className="input"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Enter todo"
  type="text"
  value=""
/>
`;
